metricflow/plan_conversion/node_processor.py

```diff
@@ -332,10 +334,19 @@ def remove_unnecessary_nodes(
         relevant_nodes = []
 
         for node in nodes:
+            logger.debug(f"Examining {node} for pruning")
             data_set = self._node_data_set_resolver.get_output_data_set(node)
             element_names_in_data_set = ToElementNameSet().transform(data_set.instance_set.spec_set)
+            element_names_intersection = element_names_in_data_set.intersection(relevant_element_names)
+            if len(element_names_intersection) > 0:
+                logger.debug(f"Including {node} since `element_names_intersection` is {element_names_intersection}")
+                relevant_nodes.append(node)
+                continue
 
-            if len(element_names_in_data_set.intersection(relevant_element_names)) > 0:
+            # Used for group-by-item-values queries.
+            if node == time_spine_node:
```

Comment:
Also a behavior change - does this impact the output?

Reply:
This does not as the node it won't be selected unless required.